Upgrade API

JacobMGEvans opened the pull request:
copilot:all

JacobMGEvans committed:
Upgrade API

nobrayner reviewed (dismissed):
Looks about right. Lets see what blows up I guess :joy:

JacobMGEvans committed:
Removing Firebase -- Rewrite to Prisma & PlanetScale

JacobMGEvans commented:
> Looks about right. Lets see what blows up I guess 😂

AHAHAHAHA That was only the start

JacobMGEvans committed:
Remove FireStore

JacobMGEvans committed:
Prisma, PlanetScale & data transforms to seed PS DB

JacobMGEvans committed:
raid ids will be dymanically aquired and used as a mappable accessor

nobrayner commented on schema.prisma:
```diff
@@ -0,0 +1,41 @@
+    avatarUrl       String
+    deletions       Int
+    additions       Int
+    raidStatsId     String?
+    RaidStats       RaidStats? @relation(fields: [raidStatsRaidId], references: [raidId])
+    raidStatsRaidId String?
```
I don't really get what is going on here... Is a Contributor tied to a single raidstat?

JacobMGEvans replied:
Yes, RaidStat is a Raid, Contributors are all the people on a Raid

JacobMGEvans committed:
ts -> js for node reasons

JacobMGEvans committed:
transformation not correct

JacobMGEvans committed:
improving script

JacobMGEvans committed:
new migration

JacobMGEvans committed:
Go Ham

JacobMGEvans committed:
dumb pipes and pretend typescript

JacobMGEvans committed:
Script Overhaul
- Refactored heavily the original script
- Added TypeScript to improve type safe data migration
- Updating Model to reflect more data elements from the documents

Co-authored-by: Theo Browne <me@t3.gg>

JacobMGEvans committed:
Loading stuff working

JacobMGEvans committed:
remove unique

JacobMGEvans committed:
save point

nobrayner reviewed (commented):
Looking great 🙌

JacobMGEvans committed:
Contributors Added to Upload

JacobMGEvans committed:
Model Changes

JacobMGEvans committed:
Contributors loading into Table

JacobMGEvans committed:
Loads RaidContributions & Updated types to handle files

JacobMGEvans committed:
model and script updates

JacobMGEvans committed:
updates

nobrayner reviewed (commented):
:+1:

JacobMGEvans commented:
copilot:all
